server/blob/blob-cassandra/src/test/java/org/apache/james/blob/cassandra/CassandraBlobsDAOTest.java

```diff
@@ -88,9 +89,24 @@ public void readShouldReturnSplitSavedDataByChunk() throws IOException {
         String longString = Strings.repeat("0123456789\n", MULTIPLE_CHUNK_SIZE);
         BlobId blobId = testee.save(longString.getBytes(StandardCharsets.UTF_8)).join();
 
-        byte[] bytes = testee.read(blobId).join();
+        byte[] bytes = testee.readBytes(blobId).join();
 
         assertThat(new String(bytes, StandardCharsets.UTF_8)).isEqualTo(longString);
     }
 
+    @Test
+    public void readBytesShouldEventualyReturnSameDataAsRead() throws IOException {
```

Comment:
Because that wasn't the point of this test. the last assertion should have been removed  as it is already tested in the previous test. The point here is to ensure that the bytes returned by both apis are the same.

Reply:
> The point here is to ensure that the bytes returned by both apis are the same.

Testing A and B are equal to C covers A equals B so I don't really understand why we need to check the two `read` methods against each others if we check both method against expected result.

Did I missed something ?

Reply:
I think that in this case (save then read) we value much more the fact that the retrieved data is the one we put in than anything else.
Having both APIs returning the same thing is obviously an interesting property but it's clearly a consequence of data not being corrupt.
Comparing read* outputs doesn't prevent the data from being corrupted, right ?

Reply:
> I think that in this case (save then read) we value much more the fact that the retrieved data is the one we put in than anything else.
> Comparing read* outputs doesn't prevent the data from being corrupted, right ?
I duplicated the sanity save/read tests in ObjectStoreContract for both methods in 0232597

I can delete the equivalence test if that's not interesting to have.

Reply:
done (test deleted)